src/main/java/com/celements/store/CelHibernateStore.java

```diff
@@ -0,0 +1,101 @@
+package com.celements.store;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.ImmutableDocumentReference;
+
+import com.google.common.collect.Iterables;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.store.XWikiHibernateStore;
+
+@Singleton
+@Component
+public class CelHibernateStore extends XWikiHibernateStore {
+
+  @Override
+  public void saveXWikiDoc(XWikiDocument doc, final XWikiContext context,
+      final boolean bTransaction) throws XWikiException {
+    // XWikiHibernateStore.saveXWikiDoc requires a mutable docRef
+    DocRefMutabilityExecutor<Void> exec = new DocRefMutabilityExecutor<Void>() {
+
+      @Override
+      public Void call(XWikiDocument doc) throws XWikiException {
+        CelHibernateStore.super.saveXWikiDoc(doc, context, bTransaction);
+        return null;
+      }
+    };
+    exec.execute(doc);
+  }
+
+  @Override
+  public XWikiDocument loadXWikiDoc(XWikiDocument doc, final XWikiContext context)
+      throws XWikiException {
+    // XWikiHibernateStore.loadXWikiDoc requires a mutable docRef
+    DocRefMutabilityExecutor<XWikiDocument> exec = new DocRefMutabilityExecutor<XWikiDocument>() {
+
+      @Override
+      public XWikiDocument call(XWikiDocument doc) throws XWikiException {
+        return CelHibernateStore.super.loadXWikiDoc(doc, context);

```

Comment:
and what about the docRefs contained in objects? would those be immutable as well after loadXWikiDoc/saveXWikiDoc? it is not obvious to me since I find those methods extremely hard to understand. I think it it not the scope of this PR, so I suggest the follow up issues:
1. copy loadXWikiDoc/saveXWikiDoc from XWikiHibernateStore and refactor to achieve
   - improved readability
   - immutable references so make DocRefMutabilityExecutor obsolete
1. improve performance by reducing amount of hqls executed